openedx/core/djangoapps/content/search/documents.py

```diff
@@ -162,7 +162,7 @@ def _tags_for_content_object(object_id: UsageKey | LearningContextKey) -> dict:
     # tags for each component separately.
     all_tags = tagging_api.get_object_tags(object_id).all()
     if not all_tags:
-        return {}
+        return {Fields.tags: {}}
```

Comment:
This was needed to clear out tags in the index when unselecting all tags for the block, it would remain the last value otherwise because the `Fields.tags` value would not be included.

Reply:
Might be worth mentioning that in the code as a comment :)